### ISSUE TITLE
Available under other another license than GPLv3 ?

### ISSUE THREAD

**Issue by henrywood** [mridgers/clink#418]
Hi

I found clink while working on a commercial application (that will be distributed) and seeing it is GPL v3, I understand that I am then obliged to make the source of our commercial program available under the same license - However, this is not an option for my employer.

Therefore, I wanted to ask if clink is available under another license.

The reason why I need it is simply to get command line history (for auditing purposes) and clink has this feature. (I am going to use clink as-is if I am allowed to do so under another license).

Thanks in advance.

Kind regards,

Henrik Skov

**Comment by koppor** [comment]
I'm not sure how you are interacting with clink. If you are using the command line parameters, I think, you need not to license your whole program under GPL. Example: If you use grep ([license: GPL](https://www.gnu.org/software/grep/)) in your scripts, the scripts need to be open source. If you use the API, it might happen that you have to license the calling code under GPL. THis is a case-by-case decision. See also the discussion about user-space drivers using the API of the GPLed Linux kernel: https://liquidat.wordpress.com/2007/07/21/linux-kernel-2623-to-have-stable-userspace-driver-api/. (**No legal advice, not confirmed, just giving ideas to start investigating other aspects.**)

In case your interpretation of the GPL stays the same, I would recommend to add dual-license to clink. For instance, the [Eclipse Public License](https://tldrlegal.com/license/eclipse-public-license-1.0-(epl-1.0)) is a good license to start with. It is commercially accepted (IBM, Bosch, etc.), but also allows usage in commercial applications without disclosure of the source.

The JGraphT project switched from LGPL to LGPL/EPL dual licensing to be more commercial friendly: https://github.com/jgrapht/jgrapht/wiki/Relicensing

**Comment by dschwoerer** [comment]
Not sure, but does clink not linking against GNU readline?
If so, you must have a GPLv3 compatible license. I don't see how changing clink would help - as it still comes with GNU readline, which still requires that the software stays free.

**Comment by chrisant996** [comment]
Correct.  Changing the clink license is not possible without removing the dependency on GNU readline, which is kind of central to what clink is.
